astropy/table/tests/test_table.py

```diff
@@ -849,6 +849,27 @@ def test_remove_row_preserves_meta(self, table_types):
         assert self.t.dtype == np.dtype([(str('a'), 'int'),
                                          (str('b'), 'int')])
 
+    def test_delitem_row(self, table_types):
+        self._setup(table_types)
+        self.t.add_column(self.b)
+        self.t.add_column(self.c)
+        del self.t[1]
+        assert self.t.colnames == ['a', 'b', 'c']
+        assert np.all(self.t['a'] == np.array([1, 3]))
+
+    def test_delitem_row_list(self, table_types):
+        self._setup(table_types)
+        self.t.add_column(self.b)
+        self.t.add_column(self.c)
+        del self.t[[0, 2]]
+        assert self.t.colnames == ['a', 'b', 'c']
+        assert np.all(self.t['c'] == np.array([8]))
+
+    def test_delitem_row_fail(self, table_types):
+        self._setup(table_types)
+        with pytest.raises(IndexError):
```

Comment:
For completeness, also add a test here where you put in something that is not an integer, e.g., `del self.t[1.]`